### DISCUSSION TITLE
Start up error

### DISCUSSION THREAD

**Discussion by tinman0608** [joeyagreco/leeger#113] [Q&A] [answered]
First, this seems awesome and I'm excited to get it working. I'm having an issue with the League loader. I've got a yahoo fantasy league. I downloaded the yahooleagueloaderexample.py sample code and changed the values for client_id and client_secret to my app on the yahoo developer site. When I run the script, I get this error.

FileNotFoundError: [WinError 2] The system cannot find the file specified
Traceback (most recent call last):
  File "C:\Users\mcken\yahooleagueloaderexample.py", line 17, in <module>
    league: League = yahooLeagueLoader.loadLeague()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mcken\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\leeger\league_loader\YahooLeagueLoader.py", line 130, in loadLeague
    yahooLeagues = self.__getAllLeagues()
                   ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\mcken\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\leeger\league_loader\YahooLeagueLoader.py", line 88, in __getAllLeagues
    yahooContext = YahooContext()
                   ^^^^^^^^^^^^^^
  File "C:\Users\mcken\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\yahoofantasy\context.py", line 30, in __init__
    raise ValueError(
ValueError: Client ID, secret, and refresh token are required. Did you run 'yahoofantasy login' already?

**Comment by joeyagreco** [comment]
Hey @tinman0608 thanks for checking out the library!

Could you send the full code that you're running (make sure to fill your client secret / client id with fake values)? 

This will help me find the issue the quickest.

**Reply by tinman0608** [reply]
Here's the script I'm running with the client id/secret changed
[fantasyfootball.txt](https://github.com/joeyagreco/leeger/files/13323961/fantasyfootball.txt)

**Reply by joeyagreco** [reply]
Ah I see the issue. 

Make sure here

```python
yahooLeagueLoader = YahooLeagueLoader(
    "123456", [2022, 2023], clientId=clientId, clientSecret=clientSecret
)
```

That you put YOUR league ID and the years for that league that you want to load.

So say you want to load 2020 and 2021, you would give the League ID for 2021 and put 2020 and 2021 for the years.

Give the most recent League ID (Based off the years you want) and give all years in order that you want.

The steps to find Yahoo League ID can be found [here](https://help.yahoo.com/kb/fantasy-football/find-league-group-number-sln8238.html).

**Reply by tinman0608** [reply]
Ok, I found my league ID and ran the script with it and I'm getting the same error (attached updated script with edited league ID and client info)
[fantasyfootball.txt](https://github.com/joeyagreco/leeger/files/13326117/fantasyfootball.txt)
. The notes say it should generate a browser window for the yahoo OAuth, but I never see one generated. Which may make sense given the error message I'm getting. Edge is my default browser, I made sure to allow all pop ups.

**Reply by joeyagreco** [reply]
Ok interesting. So the issue is that you _should_ be seeing a browser window pop up asking you to allow the login but you aren't.

That explains the error.

Did you make the callback/redirect URL this when you set up your application in the Yahoo registry? `https://localhost:8000`

I do wonder if Edge has something to do with it.

If you're certain you've followed these steps correctly

```python
    # You will need an application registered on the Yahoo Developer Site: https://developer.yahoo.com/apps/.
    # You will need your client ID and secret.
    # The app only needs to have read permissions.
    # The app will need to have a callback/redirect URI of https://localhost:8000
```

Then maybe try making your default browser Chrome/Firefox and see if that works.

You could always manually bypass this (I think) by navigating to the directory that your script is in (in a terminal) and running `yahoofantasy login`. This _should_ open up the browser window and ask you to allow.

**Comment by tinman0608** [comment, marked as answer]
So after some troubleshooting, I figured it out and... I'm dumb. My script wasn't in the same directory as everything else so my client info wasn't getting passed through correctly. Thanks for the help!
